### PR TITLE
Filter deprecated cookbooks from cookbooks list

### PR DESCRIPTION
**Pull request by manick-vel-11**
### Description

This change is related to filtering out deprecated cookbooks from cookbooks list. Previously, both deprecated and undeprecated ones will be displayed, now deprecated ones will be removed from the list.

### Issues Resolved

https://github.com/chef/supermarket/issues/2285

### Check List

- [x] New functionality includes tests
- [x] All buildkite tests pass
- [x] Full omnibus build and tests in buildkite pass
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/main/CONTRIBUTING.md#developer-certification-of-origin-dco>
- [ ] PR title is a worthy inclusion in the CHANGELOG

### CODE REVIEW

**Comment by manick-vel-11** [comment]
All cookbooks:
all_cookbooks_dep_undep


Removing deprecated cookbooks:
undep_cookbooks

**Comment by dheerajd-msys** [comment]
Looks good!! Could you please attach adhoc build @manick-vel-11 ?

**Comment by msys-sgarg** [comment]
Looks good manick

**Comment by tas50** [comment]
I think we're going to need to make it so that users can still find these deprecated cookbooks if they choose. otherwise, they're just lost forever. You'll want to want to add a search option for this.

Something like this where the deprecated cookbook option is unchecked by default

image

File: src/supermarket/app/controllers/cookbooks_controller.rb [hunk 1]
```diff
@@ -31,7 +31,12 @@ def index
       @cookbooks = @cookbooks.featured
     end
 
-    if @current_params[:order].present?
+    if (order = @current_params[:order]).present?
+
+      if order == "recently_updated"
+        @cookbooks = @cookbooks.undeprecated
```
Comment (btm) [inline src/supermarket/app/controllers/cookbooks_controller.rb:37]: "undeprecated" sounds wrong. A cookbook is either deprecated, or it is not deprecated.

Can we `@cookbooks = @cookbooks - @cookbooks.deprecated` ?

Or change the scope to something like: `scope :not_deprecated, -> { where.not(id: deprecated) }` ?

**Comment by msys-sgarg** [comment]
Please add the view specs also

**Comment by manick-vel-11** [comment]
By default, only the not deprecated cookbooks is shown.
not_deprecated_cookbooks


If include deprecated cookbooks is checked, both the deprecated and not deprecated cookbooks will be shown.
include_deprecated_cookbooks

**Review by saghoshprogress — APPROVED** [review]

**Review by dheerajd-msys — APPROVED** [review]

**Pull request merged**